src/overlays/actors/ovl_En_Kitan/z_en_kitan.c

```diff
@@ -27,50 +30,364 @@ ActorInit En_Kitan_InitVars = {
+                    break;
+
+                case 0x04B4:
+                    // Won the quiz
+                    Message_CloseTextbox(play);
+                    this->actionFunc = EnKitan_OfferPrize;
+                    EnKitan_OfferPrize(this, play);
+                    break;
+
+                case 0x04B3:
+                    // Answered a question incorrectly, stop
+                    SEQCMD_STOP_SEQUENCE(SEQ_PLAYER_FANFARE, 0);
+                    FALLTHROUGH;
+                case 0x04B5:
+                    // Keaton leaving
+                    Message_CloseTextbox(play);
+                    this->actionFunc = EnKitan_Leave;
+                    this->timer = 4;
+                    EnKitan_SpawnEffects(this, play, 30);
+                    SoundSource_PlaySfxAtFixedWorldPos(play, &this->actor.world.pos, 30, NA_SE_EN_NPC_FADEAWAY);
+                    Flags_SetCollectible(play, ENKITAN_GET_COLLECT_FLAG(&this->actor));
+                    break;
 
-extern ColliderCylinderInit D_80C09D50;
+                default:
+                    if (!(play->msgCtx.currentTextId & 1)) {
+                        // Even-numbered textboxes are question textboxes
+                        // The following textbox contains the associated answer choices for this question
+                        Message_ContinueTextbox(play, play->msgCtx.currentTextId + 1);
+                    }
+                    break;
+            }
+            break;
 
-extern UNK_TYPE D_06000CE8;
-extern UNK_TYPE D_06002770;
+        default:
+            break;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kitan/EnKitan_Init.s")
+void EnKitan_WaitForPlayer(EnKitan* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kitan/EnKitan_Destroy.s")
+    if (Actor_TalkOfferAccepted(&this->actor, &play->state)) {
+        // Began talking
+        this->actionFunc = EnKitan_Talk;
+        Message_StartTextbox(play, 0x04B0, &this->actor);
+        this->timer = 0;
+        Animation_MorphToLoop(&this->skelAnime, &gKeatonChuckleAnim, -5.0f);
+        Audio_PlayFanfare(NA_BGM_KEATON_QUIZ);
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kitan/func_80C0923C.s")
+    if ((this->timer <= 0) || (Player_GetMask(play) != PLAYER_MASK_KEATON)) {
+        // If the player does not talk quickly enough or the player isn't wearing the keaton mask, leave
+        this->actionFunc = EnKitan_Leave;
+        this->timer = 4;
+        EnKitan_SpawnEffects(this, play, 30);
+        SoundSource_PlaySfxAtFixedWorldPos(play, &this->actor.world.pos, 30, NA_SE_EN_NPC_FADEAWAY);
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kitan/func_80C09390.s")
+    if (EnKitan_CanTalk(this, play)) {
+        // Broadcast talk request for the player to accept
+        Actor_OfferTalk(&this->actor, play, 130.0f);
```

Comment:
Probably don't need this comment as this function is already well documented at the source

Reply:
Given that the rest of this function is commented in the same way I think is fine to keep this comment